timur/lib/client/jsx/actions/magma_actions.js

```diff
@@ -397,28 +395,30 @@ const cleanFileCollectionRevisions = (revisions, model_template) => {
 // Download a TSV from magma via Timur.
 export const requestTSV = (params) => (dispatch) => getTSVForm(params)
 
-export const requestAnswer = (question, callback) => {
+export const requestAnswer = (question, callback = null) => {
   return (dispatch) => {
     let localSuccess = (response) => {
       if ('error' in response) {
         dispatch(showMessages([`There was a ${response.type} error.`]));
         console.log(response.error);
-        return;
+        if(!callback) throw new Error(response.error);
```

Comment:
Same here.  To preserve the old original behavior in other places in the code, this is conditional on having passed in a callback or expecting the promise to be returned directly.  Not great, but I think to tackle this as a greater refactor we'd need to do some planning.